include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -275,8 +278,8 @@ struct transform_reduce
     _Size
     output_size(const _Size& __n, const ::std::uint16_t& __work_group_size) const
     {
-#ifndef _ONEDPL_DETECT_SPIRV_COMPILATION
-        if constexpr (_Commutative::value)
+#if !_ONEDPL_DETECT_SPIRV_COMPILATION
```

Comment:
If I understand correctly, this `output_size` call correlates with the sequential vs non-sequential implementations.  
It seems error prone to be split like this. 

For now, I'd settle for a comment similar to above but it seems like we should be able to provide a constexpr helper which determines "sequential" vs "non-sequential" and then includes the commutative check and the `_ONEDPL_DETECT_SPIRV_COMPILATION` in one consolidated place for the `transform_reduce` struct.

Reply:
I have consolidated this in the `use_nonseq_impl` helper function.